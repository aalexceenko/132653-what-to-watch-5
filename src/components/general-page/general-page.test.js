import React from "react";
import renderer from "react-test-renderer";
import GeneralPage from "./general-page";
import {BrowserRouter} from 'react-router-dom';
jest.mock(`../promo-film/promo-film`, () => `PromoFilm`);
jest.mock(`../genres-list/genres-list`, () => `GenresList`);
jest.mock(`../film-list-container/film-list-container`, () => `FilmListContainer`);


it(`Should GeneralPage render correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter >
          <GeneralPage
            onFilmCardClick={() => {}}
            handleButtonPlayVideo={() => {}}
          />
        </BrowserRouter>

    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
