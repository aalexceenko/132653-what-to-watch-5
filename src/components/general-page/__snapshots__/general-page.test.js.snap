// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GeneralPage render correctly 1`] = `
Array [
  <div
    className="visually-hidden"
  >
    <svg
      xlinkHref="http://www.w3.org/1999/xlink"
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="add"
        viewBox="0 0 19 20"
      >
        <title>
          +
        </title>
        <desc>
          Created with Sketch.
        </desc>
        <g
          fill="none"
          fillRule="evenodd"
          id="Page-1"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            id="+"
            points="10.777832 11.2880859 10.777832 19.5527344 8.41650391 19.5527344 8.41650391 11.2880859 0.627929688 11.2880859 0.627929688 8.92675781 8.41650391 8.92675781 8.41650391 0.662109375 10.777832 0.662109375 10.777832 8.92675781 18.5664062 8.92675781 18.5664062 11.2880859"
          />
        </g>
      </symbol>
      <symbol
        id="full-screen"
        viewBox="0 0 27 27"
      >
        <path
          clipRule="evenodd"
          d="M23.8571 0H16V3.14286H23.8571V11H27V3.14286V0H23.8571Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M27 23.8571V16H23.8571V23.8571H16V27H23.8571H27L27 23.8571Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M0 3.14286L0 11H3.14286L3.14286 3.14286L11 3.14286V0H3.14286H0L0 3.14286Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M3.14286 27H11V23.8571H3.14286L3.14286 16H0L0 23.8571V27H3.14286Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="in-list"
        viewBox="0 0 18 14"
      >
        <path
          clipRule="evenodd"
          d="M2.40513 5.35353L6.1818 8.90902L15.5807 0L18 2.80485L6.18935 14L0 8.17346L2.40513 5.35353Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="pause"
        viewBox="0 0 14 21"
      >
        <symbol
          id="play-s"
          viewBox="0 0 19 19"
        >
           
          <path
            clipRule="evenodd"
            d="M0 0L19 9.5L0 19V0Z"
            fill="#EEE5B5"
            fillRule="evenodd"
          />
           
        </symbol>
        <title>
          Artboard
        </title>
        <desc>
          Created with Sketch.
        </desc>
        <g
          fill="none"
          fillRule="evenodd"
          id="Artboard"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            fillRule="nonzero"
            id="Line"
            points="0 -1.11910481e-13 4 -1.11910481e-13 4 21 0 21"
          />
          <polygon
            fill="#EEE5B5"
            fillRule="nonzero"
            id="Line"
            points="10 -1.11910481e-13 14 -1.11910481e-13 14 21 10 21"
          />
        </g>
      </symbol>
    </svg>
  </div>,
  <PromoFilm
    handleButtonPlayVideo={[Function]}
  />,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <GenresList />
      <FilmListContainer
        onFilmCardClick={[Function]}
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
